processing/src/main/java/org/apache/carbondata/processing/sortandgroupby/sortdata/IntermediateFileMerger.java

```diff
@@ -110,7 +116,11 @@ public IntermediateFileMerger(SortParameters mergerParameters, File[] intermedia
       initialize();
 
       while (hasNext()) {
-        writeDataTofile(next());
+        if (useKettle) {
```

Comment:
I think it is better move this check out of while loop, compiler may have more optimization?